core/slice.go

```diff
@@ -210,39 +195,41 @@ func (sl *Slice) Append(header *types.Header, domPendingHeader *types.Header, do
 			if err != nil {
 				return nil, err
 			}
-			time9_1 = common.PrettyDuration(time.Since(start))
+			time8_1 = common.PrettyDuration(time.Since(start))
 			// Cache the subordinate's pending ETXs
 			pEtxs := types.PendingEtxs{block.Header(), subPendingEtxs}
-			time9_2 = common.PrettyDuration(time.Since(start))
-			// Add the pending etx given by the sub in the rollup
-			sl.AddPendingEtxs(pEtxs)
+			time8_2 = common.PrettyDuration(time.Since(start))
+			if pEtxs.IsValid(trie.NewStackTrie(nil)) {
```

Comment:
if `pEtxs.IsValid()` fails we should throw an error and revert the block, right?

Reply:
or just remove the check

Reply:
I removed the check from here. It is an interesting point that we are susceptible to attacks where subs have processed something and dom processes data generated from sub append in dom and is invalid. This should never happen but this is something to think about. 

Main point to take away is that, any block processing for dom block has do be done before the sub append and write tests around functions after the sub append and make sure it never breaks.